src/components/inline-link/index.tsx

```diff
@@ -1,24 +1,27 @@
 import { ReactElement } from 'react'
 import classNames from 'classnames'
 import MaybeInternalLink from 'components/maybe-internal-link'
-import Text from 'components/text'
 import { InlineLinkProps } from './types'
 import s from './inline-link.module.css'
 
 const InlineLink = ({
   className,
   href,
-  text,
-  textSize,
-  textWeight,
+  textSize = 300,
+  textWeight = 'regular',
+  children,
+  ...rest
 }: InlineLinkProps): ReactElement => {
-  const classes = classNames(s.root, className)
+  const classes = classNames(
+    s.root,
+    className,
+    `hds-typography-body-${textSize}`,
+    `hds-font-weight-${textWeight}`
+  )
```

Comment:
🚧 Can we apply `className` last so it's easy to customize any of the CSS properties?

```suggestion
  const classes = classNames(
    s.root,
    `hds-typography-body-${textSize}`,
    `hds-font-weight-${textWeight}`
    className,
  )
```